agimus_controller/agimus_controller/ocp_base_croco.py

```diff
@@ -0,0 +1,226 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+import crocoddyl
+import mim_solvers
+import numpy as np
+import pinocchio as pin
+
+from agimus_controller.mpc_data import OCPResults, OCPDebugData
+from agimus_controller.trajectory import WeightedTrajectoryPoint
+from agimus_controller.ocp_base import OCPBase
+from agimus_controller.ocp_param_base import OCPParamsCrocoBase
+
+
+class OCPCrocoBase(OCPBase):
+    def __init__(
+        self,
+        rmodel: pin.Model,
+        cmodel: pin.GeometryModel,
+        OCPParams: OCPParamsCrocoBase,
```

Comment:
IMO `OCPParamsCrocoBase` should be simply arguments of the `__init__` function. In C++ it's required due to registers and stuff, but python doesn't care

Reply:
i believe that having a data class is better for readability / organisation purpose, and so that you don't have gazillion of stuff in the init of the class